common/datavalues/src/data_value.rs

```diff
@@ -258,6 +273,62 @@ impl DataValue {
     }
 }
 
+impl Eq for DataValue {}
+
+impl Ord for DataValue {
```

Comment:
Not sure if we can define a meaningful total order relation between `DataValue`s. 
@sundy-li any comments about this? thanks

Reply:
@sundy-li, shall we reuse the `impl<S> ChangeIf<S> for CmpMin/Max ` for the comparession of `DataValue`

https://github.com/datafuselabs/databend/blob/a5748357bdbe5e5ee694ac6c1d750d1b0fe78e3e/common/functions/src/aggregates/aggregate_scalar_state.rs#L45-L73

Reply:
`aggregate_scalar_state` is generic comparision impls without matching each `DataValue`, so it's optimized by performance.


In `cluster key`, we already have min/max indexes, so we don't need to copy this indexes into a Column and apply `eval_aggr`.

Reply:
> `aggregate_scalar_state` is generic comparision impls without matching each `DataValue`, so it's optimized by performance...

got it.

Two other concerns:
1.  Although we can define our own total order for DataValue (even for weird things like NaN...)
     but a PartialOrder may be easier to define (or just use rust's derive mechanism)
2. The min/max logic is also implemented by the aggregate_min/max
     It would be better if these two keep in coherent.     

A suggestion:

- derive `PartialOrd` for DataValue      
- implements the min/max like the `aggregate_scalar_state` does 
   maybe something like this (hope it is doing the same thing as aggregate min max does:-) 
   
   ~~~
           let min = min_stats
                 .iter()
                 .filter(|x| !x.is_null())
                 .min_by(|x, y| min.partial_cmp(&r).unwrap_or(Ordering::Equal))
                 .unwrap_or(DataValue::Null);

            let max = max_stats
                 .iter()
                 .filter(|x| !x.is_null())
                 .min_by(|x, y| max.partial_cmp(&r).unwrap_or(Ordering::Equal))
                 .unwrap_or(DataValue::Null);
~~~

Reply:
The logics want to follow:

https://github.com/datafuselabs/databend/blob/6623a328b70d793c6f00bb97807ff9df2d4fc1f2/common/functions/src/aggregates/aggregate_scalar_state.rs#L126-L135

https://github.com/datafuselabs/databend/blob/6623a328b70d793c6f00bb97807ff9df2d4fc1f2/common/functions/src/aggregates/aggregate_scalar_state.rs#L67-L71